### ISSUE TITLE
Update to DivestOS_20 2024 July 11 failing on H872

### ISSUE THREAD

**Issue by CueHD** [Divested-Mobile/DivestOS-Build#321]
When attempting to update from the System Updater within Settings, the updating fails while in recovery. The screen shows "error" under the Android logo.

I tried to update from within DivestOS recovery. The error shown is

ERROR: recovery: Error exit from the fuse process: 1
[Install completed with status 1.
Installation aborted

Also, I was unable to find within the DivestOS recovery an option to update from SD card. Is it not available?

**Comment by SkewedZeppelin** [comment]
Can you please download the update to a computer and manually apply the update via adb sideload?

**Comment by CueHD** [comment]
I can't seem to get my computer to recognize the phone at all. I have tried 3 different computers (Windows 10, 11, Linux Mint 20), several USB cables and ports on 2 of the computers (one laptop only has one port).

The settings app on the phone has many of the USB options greyed out. They only one I seem able to change is the "default USB connection" within developer settings. I am able to charge rapidly from an AC/DC adapter but the ports on the computer do not. "ADB devices" does not show a connection.

Is there a setting or restriction that I am missing which may be restricting my options?

**Comment by SkewedZeppelin** [comment]
it is possible, but unlikely that I broke USB
however to sideload the update you need to be in recovery mode, and that'd also rule out other OS aspects

**Comment by CueHD** [comment]
Yes, I initially tried to connect while in recovery and the phone was not seen.  I did not try all of the various cables, ports etc. from recovery, so I can try again. Is it possible that something regarding USB broke in the recovery partition? I wouldn't know if there has been any change since every update has succeeded from the updater.

I could try installing a different recovery from the DivestOS recovery and see if it changes anything, but I'll try connecting USB again  first.

**Comment by steadfasterX** [comment]
There's also mAid (https://maid.binbash.rocks) for ruling out OS issues

**Comment by CueHD** [comment]
I believe the SkewedZepplin was referring to DivestOS when he referred to "OS issues". Thank you for the link, but I know that two of the computers I tried have everything they need to connect to that phone as I used them for unlocking the bootloader and initial installation of custom recovery.

I was able to get a connection using a new cable and USB-C to USB-A adapter while DivestOS system was loaded, but then it did not connect while in DivestOS recovery. I wonder if my phone's USB port is having problems. I'll continue to try later today.

**Comment by CueHD** [comment]
Is there something about the broken LineageOS July builds that is causing DivestOS issues too?

https://xdaforums.com/t/official-lineageos-21-for-the-lg-g6.4649860/post-89597481

I realize the link is about an unofficial LineageOS21 that is not directly related to the official DivestOS20.

**Comment by SkewedZeppelin** [comment]
LineageOS 20.0 is largely frozen.

You can have a USB port that charges but doesn't allow data, I have quite a few test devices with bad ports like that.

**Comment by CueHD** [comment]
I hope it's not the USB port. Is there a way to connect ADB through WiFi while in DivestOS recovery? If not, I suppose I'll have to wait until next month's update and hope that one can be installed.

**Comment by SkewedZeppelin** [comment]
if usb does not work recovery and fastboot can't really be used

you can try deleting the update in the Updater app and downloading it again
also make sure you have a few GB of free space available

**Comment by CueHD** [comment]
I have ≈ 15 GB free on the device and ≈ 28 GB free on the SD card.

Is there a way to verify the sha512 on Android? I haven't been able to find an app that can.

**Comment by CueHD** [comment]
I am able to access fastboot from bootloader and ADB from system but not ADB from recovery.

I assume that I should try flashing a different recovery and go from there. Would you recommend that as the next step? Did anything in the recovery change from the May to the June version?

**Comment by CueHD** [comment]
I changed to OrangeFox recovery using Fastboot and was able to update that way. When I updated, the recovery was updated to the latest DivestOS recovery. I will reply to this issue when the October update is attempted.

**Comment by CueHD** [comment]
The DivestOS updater did not work again. I'll likely stick with OrangeFox once I get it back on my H872. If you make any changes that might fix the issue, let me know, and I will test if things change along with them.
